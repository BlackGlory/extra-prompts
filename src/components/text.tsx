import { Box, Text as InkText } from 'ink'
import { TextInput } from '@inkjs/ui'

interface ITextProps {
  message: string
  onSubmit(value: string): void
}

export function Text(props: ITextProps) {
  const { message, onSubmit } = props

  return (
    <Box>
      <InkText>{message}: </InkText>
      <TextInput onSubmit={onSubmit} />
    </Box>
  )
}
