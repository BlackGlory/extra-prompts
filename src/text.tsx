import { Text } from '@components/text'
import { render } from 'ink'

export function text(message: string): Promise<string> {
  return new Promise(resolve => {
    const { unmount } = render(
      <Text
        message={message}
        onSubmit={value => {
          unmount()
          resolve(value)
        }}
      />
    )
  })
}
