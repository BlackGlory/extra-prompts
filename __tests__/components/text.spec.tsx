import { Text } from '@components/text.js'
import { render } from 'ink-testing-library'
import { delay } from 'extra-promise'

describe('Text', () => {
  test('render', () => {
    const fn = vi.fn()
    const { lastFrame } = render(
      <Text
        message='message'
        onSubmit={fn}
      />
    )

    expect(lastFrame()).toMatchSnapshot()
    expect(fn).toBeCalledTimes(0)
  })

  test('onSubmit', async () => {
    const fn = vi.fn()
    const { stdin, lastFrame } = render(
      <Text
        message='message'
        onSubmit={fn}
      />
    )

    await delay(100)
    stdin.write('hello')
    await delay(100)
    stdin.write('\r')
    await delay(100)

    expect(lastFrame()).toMatchSnapshot()
    expect(fn).toBeCalledTimes(1)
    expect(fn).toBeCalledWith('hello')
  })
})
