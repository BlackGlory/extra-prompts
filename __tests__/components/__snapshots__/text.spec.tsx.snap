// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Text > onSubmit 1`] = `"message: hello[7m [27m"`;

exports[`Text > render 1`] = `"message: [7m [27m"`;
